src/blk/kernel/KernelDevice.cc

```diff
@@ -790,6 +790,9 @@ void KernelDevice::aio_submit(IOContext *ioc)
     return;
   }
 
+  // avoid aios being truncated when calling submit_batch.
+  assert(ioc->num_pending.load() <= 65535);
+
```

Comment:
```diff
diff --git a/src/blk/kernel/KernelDevice.cc b/src/blk/kernel/KernelDevice.cc
index e3ef9f155d8..55321af1b56 100644
--- a/src/blk/kernel/KernelDevice.cc
+++ b/src/blk/kernel/KernelDevice.cc
@@ -12,6 +12,7 @@
  *
  */

+#include <limits>
 #include <unistd.h>
 #include <stdlib.h>
 #include <sys/types.h>
@@ -813,6 +814,8 @@ void KernelDevice::aio_submit(IOContext *ioc)

   void *priv = static_cast<void*>(ioc);
   int r, retries = 0;
+  // num of pending aios should not overflow when passed to submit_batch()
+  assert(pending <= std::limits<uint16_t>::max());
   r = io_queue->submit_batch(ioc->running_aios.begin(), e,
                 pending, priv, &retries);
```

i'd suggest use the patch above for following reasons.

- avoid using magic number
- we do not "avoid" this error here. what we are doing is just to **verify** the number of pending ios does not overflow
- move the check closer to where `pending` is consumed.

Reply:
Thanks for your suggestion, I have modified it.